src/scanspec/core.py

```diff
@@ -458,7 +478,21 @@ def concat_gap(gaps: Sequence[GapArray]) -> GapArray:
             g[len(self)] = gap or gap_between_frames(self, other)
             return g
 
-        return _merge_frames(self, other, dict_merge=concat_dict, gap_merge=concat_gap)
+        def concat_duration(
+            durations: Sequence[DurationArray | None],
+        ) -> DurationArray | None:
+            # Check if there are more than one duration being zipped
+            specified_durations = [d for d in durations if d is not None]
+            d = np.concatenate(specified_durations)
+            return d
```

Comment:
I think we should require they are all not None, it doesn't make sense to concatentate them otherwise

```suggestion
            # We will be passed a sequence of durations where at least one
            # is not None. We require that they are all not None
            if any(d is None for d in durations):
                raise ValueError("Can't concatenate dimensions unless all or none provide durations")
            return np.concatenate(durations)
```